src/server/scripts/EasternKingdoms/ZulGurub/boss_thekal.cpp

```diff
@@ -84,8 +89,11 @@ class boss_thekal : public CreatureScript
             _Reset();
             Initialize();
 
+            _scheduler.CancelAll();
```

Comment:
consider inheriting BossAI instead, so you can just inherit an already created taskscheduler from there, as well as a bunch of helper methods. this will clean up the code significantly

Reply:
Am I not already inherited BossAI? Or am I doing something wrong?

https://github.com/kjack9/azerothcore-wotlk/blob/84479511e4e47180a705b659fe2b55ff35d81d95/src/server/scripts/EasternKingdoms/ZulGurub/boss_thekal.cpp#L72

Reply:
I am sorry. I worded that wrong. Inherit taskscheduler from bossai is what I meant

Reply:
Done!